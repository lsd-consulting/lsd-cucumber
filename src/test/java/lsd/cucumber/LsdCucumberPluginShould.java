package lsd.cucumber;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static java.nio.charset.Charset.defaultCharset;
import static java.nio.file.Files.readString;
import static java.nio.file.Paths.get;

class LsdCucumberPluginShould {

    @Test
    void generateMenu() throws IOException {
        Approvals.verifyHtml(readString(get("build/reports/lsd/lsd-index.html"), defaultCharset()));
    }

    @Test
    void generateReportForBasicScenario() throws IOException {
        Approvals.verifyHtml(readString(get("build/reports/lsd/Scenario.html"), defaultCharset()));
    }

    @Test
    void generateReportForScenarioWithDataTable() throws IOException {
        Approvals.verifyHtml(readString(get("build/reports/lsd/Data_table.html"), defaultCharset()));
    }

    @Test
    void generateForScenarioOutline() throws IOException {
        Approvals.verifyHtml(readString(get("build/reports/lsd/Scenario_outline.html"), defaultCharset()));
    }
}
