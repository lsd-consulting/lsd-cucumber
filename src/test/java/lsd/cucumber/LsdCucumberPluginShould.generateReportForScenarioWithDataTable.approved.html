<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns="http://www.w3.org/1999/html" xml:lang="en" lang="en">
<head>
    <title>data_table</title>
    <link rel="stylesheet" href="style.css">
</head>
<body>
<main role="main">
    <span onclick="location.href='lsd-index.html';" title="back to index page" class="back_to_index"></span>
    <h1 class="logo">LSD</h1>
    <h1>data_table</h1>
    <div class="contents">
        <h3>Contents</h3>
        <ul class="contents">
            
            <li class="success"><a href="#1">Scenario with a data table</a></li>
            
        </ul>
    </div>
    

    <article id="1">
        <span onclick="location.href='#top';" title="top of page" class="back_to_top"></span>
        <h2 class="success">Scenario with a data table</h2>
        <section class="description success">
            <h3 class="success">Description</h3>
            Given  given2</br>When  when2</br>Then  then2
        </section>
        
        
        <section class="sequence diagram svg success">
            <h3 class="success">Sequence Diagram</h3>
            <figure><?xml version="1.0" encoding="UTF-8"?>

<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" height="152px" preserveAspectRatio="none" style="width:212px;height:152px;background:#00000000;" version="1.1" viewBox="0 0 212 152" width="212px" zoomAndPan="magnify">
  <defs/>
  <g>
    <line style="stroke:#000000;stroke-width:1.0;stroke-dasharray:5.0,5.0;" x1="51" x2="51" y1="37.0146" y2="116.6133"/>
    <line style="stroke:#000000;stroke-width:1.0;stroke-dasharray:5.0,5.0;" x1="159" x2="159" y1="37.0146" y2="116.6133"/>
    <rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="62" x="20" y="5"/>
    <text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="27" y="26.0752">Source</text>
    <rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="62" x="20" y="115.6133"/>
    <text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="27" y="136.6885">Source</text>
    <rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="94" x="112" y="5"/>
    <text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="119" y="26.0752">Destination</text>
    <rect fill="#FFFFFF" height="31.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="94" x="112" y="115.6133"/>
    <text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="119" y="136.6885">Destination</text>
    <polygon fill="#000000" points="147,64.814,157,68.814,147,72.814,151,68.814" style="stroke:#000000;stroke-width:1.0;"/>
    <line style="stroke:#000000;stroke-width:1.0;" x1="51" x2="153" y1="68.814" y2="68.814"/>
    <a href="#id1" target="_top" title="label1" xlink:actuate="onRequest" xlink:href="#id1" xlink:show="new" xlink:title="label1" xlink:type="simple">
      <text fill="" font-family="Verdana" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="40" x="58" y="64.0845">label1</text>
    </a>
    <polygon fill="#000000" points="62,94.6133,52,98.6133,62,102.6133,58,98.6133" style="stroke:#000000;stroke-width:1.0;"/>
    <line style="stroke:#000000;stroke-width:1.0;" x1="56" x2="158" y1="98.6133" y2="98.6133"/>
    <a href="#id2" target="_top" title="label2" xlink:actuate="onRequest" xlink:href="#id2" xlink:show="new" xlink:title="label2" xlink:type="simple">
      <text fill="" font-family="Verdana" font-size="13" lengthAdjust="spacing" text-decoration="underline" textLength="40" x="68" y="93.8838">label2</text>
    </a>
  </g>
</svg>
</figure>
            <div onclick="location.href='#!';" id="2" class="overlay">
                <div class="popup" onclick="event.stopPropagation();">
                    <h2>PlantUml markup</h2>
                    <span class="copy" title="copy" onclick="copyToClipboard('sequence_uml_code2')"></span>
                    <a class="close" href="#!">&times;</a>
                    <div class="content">
                        <pre><code id="sequence_uml_code2">@startuml

!theme plain

skinparam backgroundColor transparent

hide unlinked






Source -&gt; Destination: &lt;text fill&#x3D;&quot;&quot;&gt;[[#id1 {label1} label1]]&lt;/text&gt;

Destination -&gt; Source: &lt;text fill&#x3D;&quot;&quot;&gt;[[#id2 {label2} label2]]&lt;/text&gt;


@enduml</code></pre>
                    </div>
                </div>
            </div>
            <sub><a href="#2">[source]</a></sub>
            <sub><a href="#3">[component diagram]</a></sub>
        </section>
        <!-- Popup for component diagram -->
        <div onclick="location.href='#!';" id="3" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>Component Diagram</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre><?xml version="1.0" encoding="UTF-8"?>

<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentStyleType="text/css" height="141px" preserveAspectRatio="none" style="width:106px;height:141px;background:#00000000;" version="1.1" viewBox="0 0 106 141" width="106px" zoomAndPan="magnify">
  <defs/>
  <g>
    <g id="elem_Source">
      <rect fill="#FFFFFF" height="47.0146" rx="2.5" ry="2.5" style="stroke:#000000;stroke-width:1.0;" width="88" x="12" y="10.0146"/>
      <rect fill="#FFFFFF" height="10" style="stroke:#000000;stroke-width:1.0;" width="15" x="80" y="15.0146"/>
      <rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="78" y="17.0146"/>
      <rect fill="#FFFFFF" height="2" style="stroke:#000000;stroke-width:1.0;" width="4" x="78" y="21.0146"/>
      <text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="48" x="27" y="44.0898">Source</text>
    </g>
    <g id="elem_Destination">
      <ellipse cx="56" cy="102.0146" fill="#FFFFFF" rx="8" ry="8" style="stroke:#000000;stroke-width:1.0;"/>
      <text fill="#000000" font-family="Verdana" font-size="14" lengthAdjust="spacing" textLength="80" x="16" y="133.0898">Destination</text>
    </g>
    <path d="M49.6208,57.2715 C48.858,69.148 49.2722,83.1547 50.8634,92.7178 " fill="none" style="stroke:#000000;stroke-width:1.0;"/>
    <polygon fill="#000000" points="50.8634,92.7178,53.3319,83.1833,50.0427,87.7856,45.4404,84.4964,50.8634,92.7178" style="stroke:#000000;stroke-width:1.0;"/>
    <path d="M61.1366,92.7178 C62.7278,83.1547 63.142,69.148 62.3792,57.2715 " fill="none" style="stroke:#000000;stroke-width:1.0;"/>
    <polygon fill="#000000" points="62.3792,57.2715,58.9643,66.5094,62.6997,62.2613,66.9479,65.9967,62.3792,57.2715" style="stroke:#000000;stroke-width:1.0;"/>
  </g>
</svg>
</pre>
                </div>
                <sub><a href="#uml3">[source]</a></sub>
            </div>
        </div>
        <!-- Popup for component diagram UML via component diagram popup -->
        <div onclick="location.href='#!';" id="uml3" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>PlantUml markup</h2>
                <span class="copy" title="copy" onclick="copyToClipboard('component_uml_code3')"></span>
                <a class="close" href="#3">&times;</a><!--show previous popup on close-->
                <div class="content">
                    <pre><code id="component_uml_code3">@startuml
!pragma layout smetana
!theme plain
skinparam backgroundColor transparent



[Source] --&gt; Destination
[Destination] --&gt; Source


remove @unlinked
@enduml</code></pre>
                </div>
            </div>
        </div>
        <!-- Section containing data for sequence diagram popups -->
        
        <div onclick="location.href='#!';" id="id1" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label1</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div>
        
        <div onclick="location.href='#!';" id="id2" class="overlay">
            <div class="popup" onclick="event.stopPropagation();">
                <h2>label2</h2>
                <a class="close" href="#!">&times;</a>
                <div class="content">
                    <pre></pre>
                </div>
            </div>
        </div>
        
        
    </article>
    
</main>
</body>
<footer>
    <script type = "text/javascript" src = "https://d3js.org/d3.v4.min.js"></script>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.0.0/build/styles/default.min.css">
    <script src="custom.js"></script>
    <script language = "javascript">
        addSliderForSvgZoom();
        highlightLifelinesWhenClicked();
    </script>
    <script src="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.0.0/build/highlight.min.js"></script>
    <script>hljs.highlightAll();</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js"></script>
    <script>
        highlightKeywords();
        
            
        
        unMarkSvg();
    </script>
</footer>
</html>
